# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/vr/service/OpenVRSession.cpp
# Commit: 1d9384906287
# Full Hash: 1d93849062870a8d492eeef2150ecea174cb6223
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2019-05-26 21:12:53
# Regressor Bug: 1550422
# File Overlap Count: 1
# Description:
#   Bug 1550422 - P15. Move Skip and Once gfxPrefs to StaticPrefs. r=jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D31259
# ==============================================================================

diff -r f5ebdcca640d -r 1d9384906287 gfx/vr/service/OpenVRSession.cpp
--- a/gfx/vr/service/OpenVRSession.cpp	Sun May 26 14:29:56 2019 +0000
+++ b/gfx/vr/service/OpenVRSession.cpp	Sun May 26 14:30:14 2019 +0000
@@ -10,7 +10,7 @@
 #include "nsString.h"
 
 #include "OpenVRSession.h"
-#include "gfxPrefs.h"
+#include "mozilla/StaticPrefs.h"
 
 #if defined(XP_WIN)
 #  include <d3d11.h>
@@ -250,7 +250,7 @@
 }
 
 bool OpenVRSession::Initialize(mozilla::gfx::VRSystemState& aSystemState) {
-  if (!gfxPrefs::VREnabled() || !gfxPrefs::VROpenVREnabled()) {
+  if (!StaticPrefs::VREnabled() || !StaticPrefs::VROpenVREnabled()) {
     return false;
   }
   if (mVRSystem != nullptr) {
@@ -306,7 +306,7 @@
     return false;
   }
 
-  if (gfxPrefs::VROpenVRActionInputEnabled() && !SetupContollerActions()) {
+  if (StaticPrefs::VROpenVRActionInputEnabled() && !SetupContollerActions()) {
     return false;
   }
 
@@ -326,7 +326,7 @@
   nsCString knucklesManifest;
 
   // Getting / Generating manifest file paths.
-  if (gfxPrefs::VRProcessEnabled()) {
+  if (StaticPrefs::VRProcessEnabled()) {
     VRParent* vrParent = VRProcessChild::GetVRParent();
     nsCString output;
 
@@ -769,7 +769,7 @@
   // End of setup controller actions.
 
   // Notify the parent process these manifest files are already been recorded.
-  if (gfxPrefs::VRProcessEnabled()) {
+  if (StaticPrefs::VRProcessEnabled()) {
     NS_DispatchToMainThread(NS_NewRunnableFunction(
         "SendOpenVRControllerActionPathToParent",
         [controllerAction, viveManifest, WMRManifest, knucklesManifest]() {
@@ -1935,7 +1935,7 @@
   UpdateHeadsetPose(aSystemState);
   UpdateEyeParameters(aSystemState);
 
-  if (gfxPrefs::VROpenVRActionInputEnabled()) {
+  if (StaticPrefs::VROpenVRActionInputEnabled()) {
     EnumerateControllers(aSystemState);
 
     vr::VRActiveActionSet_t actionSet = {0};
@@ -2147,7 +2147,7 @@
    */
   OpenVRSession* self = static_cast<OpenVRSession*>(aClosure);
 
-  if (gfxPrefs::VROpenVRActionInputEnabled()) {
+  if (StaticPrefs::VROpenVRActionInputEnabled()) {
     self->UpdateHaptics();
   } else {
     self->UpdateHapticsObsolete();